lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2822,15 +2822,16 @@ ParseResult ConstantOp::parse(OpAsmParser &parser, OperationState &result) {
 
 LogicalResult ConstantOp::verify() {
   // If the result type has a bitwidth, then the attribute must match its width.
-  auto intType = getType();
+  auto intType = getType().get();
   auto width = intType.getWidthOrSentinel();
   if (width != -1 && (int)getValue().getBitWidth() != width)
     return emitError(
         "firrtl.constant attribute bitwidth doesn't match return type");
 
   // The sign of the attribute's integer type must match our integer type sign.
-  auto attrType = getValueAttr().getType().cast<IntegerType>();
-  if (attrType.isSignless() || attrType.isSigned() != getType().isSigned())
+  auto attrType = firrtl::type_cast<IntegerType>(getValueAttr().getType());
```

Comment:
Hmm, wondering if pattern should be to just do `isa` on the `get()` instead of this `type_isa`, at least here?

As-is it already relies on the two paths being equivalent in order for this check to be correct, just trying to lower what folks keep in their mind to see why this is right.

Fine as-is, just a note.

Reply:
Yeah, we can use cast here as cast is applied to an attribute type, which has mlir type.